### DISCUSSION TITLE
Compile LSPosed fail

### DISCUSSION THREAD

**Discussion by Kev1nKe** [LSPosed/LSPosed#2757] [Q&A] [answered]
LSPosed 1.9.1 编译错误

错误一：assembleRiru
FAILURE: Build completed with 27 failures.

1: Task failed with an exception.
-----------
* What went wrong:
Execution failed for task ':core:compileDebugJavaWithJavac'.
> Could not resolve all files for configuration ':core:debugCompileClasspath'.
   > Could not find io.github.libxposed:api:100.
     Required by:
         project :core

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.
==============================================================================

错误二：assembleBuild

Could not determine the dependencies of task ':core:minifyReleaseWithR8'.
> Cannot query the value of this property because it has no value available.

**Comment by tianjyan** [comment, marked as answer]
I think you should clone libxposed/api and libxposed/service first. See the workflow in https://github.com/LSPosed/LSPosed/blob/master/.github/workflows/core.yml

**Reply by Kev1nKe** [reply]
Okay I will try thank you.

**Comment by AndroidMaster25** [comment]
Here is how to fix `Could not find io.github.libxposed:api:100` by cloning

```
git clone --depth 1 https://github.com/libxposed/api.git libxposed/api
git clone --depth 1 https://github.com/libxposed/service.git libxposed/service
 
cd libxposed/api
./gradlew :api:publishApiPublicationToMavenLocal
cd ../service
./gradlew :interface:publishInterfacePublicationToMavenLocal
```

Source: https://zwc365.com/2024/01/23/lspatch%E8%87%AA%E8%A1%8C%E7%BC%96%E8%AF%91

**Comment by Kev1nKe** [comment]
thank you solved by your answer already。



---- Replied Message ----
| From | ***@***.***> |
| Date | 01/16/2025 19:43 |
| To | ***@***.***> |
| Cc | ***@***.***>***@***.***> |
| Subject | Re: [LSPosed/LSPosed] Compile LSPosed fail (Discussion #2757) |

Here is how to fix Could not find io.github.libxposed:api:100 by cloning

git clone --depth 1 https://github.com/libxposed/api.git libxposed/api
git clone --depth 1 https://github.com/libxposed/service.git libxposed/service
 
cd libxposed/api
./gradlew :api:publishApiPublicationToMavenLocal
cd ../service
./gradlew :interface:publishInterfacePublicationToMavenLocal


Source: https://zwc365.com/2024/01/23/lspatch%E8%87%AA%E8%A1%8C%E7%BC%96%E8%AF%91
